mteb/tasks/Clustering/fra/HALClusteringS2S.py

```diff
@@ -46,3 +50,64 @@ def dataset_transform(self):
             "labels": [split.tolist() for split in np.array_split(domains, 10)],
         }
         self.dataset["test"] = datasets.Dataset.from_dict(new_format)
+
+
+class HALClusteringS2SFast(AbsTaskClusteringFast):
+    metadata = TaskMetadata(
+        name="HALClusteringS2SFast",
+        description="Clustering of titles from HAL (https://huggingface.co/datasets/lyon-nlp/clustering-hal-s2s)",
+        reference="https://huggingface.co/datasets/lyon-nlp/clustering-hal-s2s",
+        dataset={
+            "path": "lyon-nlp/clustering-hal-s2s",
+            "revision": "e06ebbbb123f8144bef1a5d18796f3dec9ae2915",
+        },
+        type="Clustering",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["fra-Latn"],
+        main_score="v_measure",
+        date=("2000-03-29", "2024-05-24"),
+        form=["written"],
+        domains=["Academic"],
+        task_subtypes=["Thematic clustering"],
+        license="Apache-2.0",
+        socioeconomic_status="medium",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="",
+        n_samples={"test": NUM_SAMPLES},
+        avg_character_length={"test": 86.6},
+    )
+
+    def dataset_transform(self):
+        """Convert to standard format"""
+        self.dataset["test"] = self.dataset["test"].remove_columns("hal_id")
+        self.dataset["test"] = self.dataset["test"].rename_columns(
+            {"title": "sentences", "domain": "labels"}
+        )
+        labels_count = Counter(self.dataset["test"]["labels"])
+
+        # leave classes with more than 2 samples after stratified_subsampling
```

Comment:
You mean keep classes with more than 2 samples?

Reply:
Yes